SmartDeviceLinkTests/ProtocolSpecs/SDLProtocolReceivedMessageProcessorSpec.m

```diff
@@ -0,0 +1,554 @@
+    ERROR_STATE = -1,
+};
+
+@interface SDLProtocolReceivedMessageProcessor(){}
+// State management
+@property (assign, nonatomic) ProcessorState state;
+
+// Message assembly state
+@property (strong, nonatomic) SDLProtocolHeader *header;
+@property (strong, nonatomic) NSMutableData *headerBuffer;
+@property (strong, nonatomic) NSMutableData *payloadBuffer;
+
+// Error checking
+@property (assign, nonatomic) UInt8 version;
+@property (assign, nonatomic) BOOL encrypted;
+@property (assign, nonatomic) SDLFrameType frameType;
+@property (assign, nonatomic) UInt32 dataLength;
+@property (assign, nonatomic) UInt32 dataBytesRemaining;
+@property (assign, nonatomic) SDLServiceType serviceType;
+@end
+
+QuickSpecBegin(SDLProtocolReceivedMessageProcessorSpec)
+
+describe(@"The processor", ^{
+    __block SDLProtocolReceivedMessageProcessor *testProcessor = nil;
+    __block NSMutableData *testBuffer;
+    __block NSMutableData *testHeaderBuffer;
+    
+    beforeEach(^{
+        testProcessor = [[SDLProtocolReceivedMessageProcessor alloc] init];
+        testBuffer = [NSMutableData data];
+        testHeaderBuffer = [NSMutableData data];
+    });
+    context(@"in START_STATE", ^{
+        beforeEach(^{
+            testProcessor.state = START_STATE;
+        });
+        context(@"When it recieves a byte", ^{
+            context(@"with a good version 1", ^{
+                it(@"transitions to SERVICE_TYPE_STATE", ^{
+                    Byte testByte = 0x11;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
+                });
+            });
+            context(@"with a good version 2", ^{
+                it(@"transitions to SERVICE_TYPE_STATE", ^{
+                    Byte testByte = 0x21;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
+                });
+            });
+            context(@"with a good version 3", ^{
+                it(@"transitions to SERVICE_TYPE_STATE", ^{
+                    Byte testByte = 0x31;
+                    [testBuffer appendBytes:&testByte length:1];
+                    
+                    [testProcessor processReceiveBuffer:testBuffer withMessageReadyBlock:^(SDLProtocolHeader *header, NSData *payload) {
+                        //do nothing?
+                    }];
+                    expect(@(testProcessor.state)).to(equal(SERVICE_TYPE_STATE));
```

Comment:
Ensure that `testProcessor.version` is set properly (same throughout this file when something gets set)